l2geth/core/blockchain.go

```diff
@@ -1452,6 +1453,11 @@ func (bc *BlockChain) writeBlockWithState(block *types.Block, receipts []*types.
 	} else {
 		bc.chainSideFeed.Send(ChainSideEvent{Block: block})
 	}
+	if bc.updateSyncService != nil {
+		if block.Transactions().Len() > 0 {
+			bc.updateSyncService(block.Transactions()[0])
```

Comment:
Do we need to slip `UpdateSyncServiceState` into two parts. The part one do precheck, and return false if failed. If precheck is failed, the block insertion should be rejected. In the part two just update sync service state without return value.